### DISCUSSION TITLE
Batch variations REST endpoint doesn't update the product "in-stock" status

### DISCUSSION THREAD

**Discussion by rudtek** [woocommerce/woocommerce#53134] [General]
I'm using the REST API to create products that have 40-80 variations each.  My best understanding is that I need to create 2 API Calls for this.  The first is to create the product and attributes using the product endpoint.  The 2nd is to use the batch variations endpoint to create the variations and attach them to the product. This is timing out for me from the remote server.

The way I'm trying is to create a product with the first endpoint and add name and details.  This leaves the product out of stock (OOS) as I haven't added pricing information.  

The next step is to add the variations. I have this working to successfully add the variations.  The problem is that it doesn't set the product to "in-stock" until I go into the WP dashboard, open the product edit screen, and hit "update".

Is there a way to do this as part of the product variation endpoint so I don't need to login to WP?

**Comment by coreymckrill** [comment]
Hi @rudtek when you're creating the variations, are you including the `in_stock` property in the request? You should be able to do something like this as one of your variation properties:

```
{
    "in_stock": true
}
```

**Reply by rudtek** [reply]
I did add it, and there is no change.  I figured it out though I believe. I'm using wpEngine as a host and they will only allow 5 seconds for a cURL timeout.  The process was timing out and not able to set the product itself (not the variations) to live.  Super frustrating.  I'll need to find an new host.  Suggestions on that?

**Reply by coreymckrill** [reply]
Here are our hosting suggestions: https://woocommerce.com/hosting-solutions/
